lib/Analysis/TargetSlotAnalysis/TargetSlotAnalysis.cpp

```diff
@@ -22,15 +24,47 @@ void TargetSlotAnalysis::visitOperation(
   llvm::TypeSwitch<Operation &>(*op)
       .Case<tensor::InsertOp>([&](auto insertOp) {
         LLVM_DEBUG({ llvm::dbgs() << "Visiting: " << *op << "\n"; });
-        auto insertIndexRes = get1DExtractionIndex<tensor::InsertOp>(insertOp);
+        auto insertIndices = insertOp.getIndices();
+        if (insertIndices.size() != 1) {
+          LLVM_DEBUG(llvm::dbgs() << "At " << insertOp
+                                  << " can't handle >1D insertion index\n");
+          return;
+        }
+
+        Value insertIndexValue = insertOp.getIndices()[0];
+        const dataflow::Lattice<dataflow::ConstantValue> *insertIndexLattice =
+            sccpAnalysis
+                ->lookupState<dataflow::Lattice<dataflow::ConstantValue>>(
+                    insertIndexValue);
+
+        if (insertIndexLattice) {
+          LLVM_DEBUG(llvm::dbgs()
+                     << "At " << insertOp << " SCCP analysis gives lattice of "
+                     << *insertIndexLattice << "\n");
+        }
```

Comment:
hey! i'm just curious, i ran into needing constant folded strides / sizes once and I came across this method: https://github.com/regehr/llvm-project/blob/ba0763e4cb08015bd0acef5cf05a372d423017b4/mlir/lib/Dialect/Vector/Transforms/VectorEmulateNarrowType.cpp#L88

can you `getAsOpFoldResult` the insert index values?

Reply:
I don't think I can, because in this case the constant analysis is simulated, not materialized in the actual IR.